test/unit/spec/MfaVerify_spec.js

```diff
@@ -3172,6 +3175,59 @@ function (Okta,
       });
     });
 
+    Expect.describe('Supports chains of MFA_REQUIRED responses', function () {
+      itp('Mfa verify follows password re-auth without trapping the auth response', function () {
+        return setupPassword().then(function (test) {
+          spyOn(RouterUtil, 'handleResponseStatus').and.callThrough();
+          $.ajax.calls.reset();
+          test.form.setPassword('Abcd1234');
+          test.form.setRememberDevice(true);
+          test.setNextResponse(resAllFactors);
+          test.form.submit();
+          return Expect.waitForVerifyQuestion(test);
+        })
+        .then(function (test) {
+          test.questionForm = new MfaVerifyForm($sandbox.find('.o-form'));
+          expect(test.questionForm.isSecurityQuestion()).toBe(true);
+          // trapAuthResponse does not prevent handleResponseStatus from being called
+          expect(RouterUtil.handleResponseStatus.calls.count()).toBe(1);
+        });
+      });
+      itp('Any factor change through the dropdown traps auth response', function () {
+        return setupSMS().then(function (test) {
+          spyOn(RouterUtil, 'handleResponseStatus').and.callThrough();
+          test.setNextResponse(resChallengeSms);
+          test.form.smsSendCode().click();
+          return tick(test);
+        })
+        .then(function (test) {
+          test.setNextResponse(resAllFactors);
+          test.beacon.dropDownButton().click();
+          clickFactorInDropdown(test, 'QUESTION');
+          return Expect.waitForVerifyQuestion(test);
+        })
+        .then(function () {
+          // trapAuthResponse prevents handleResponseStatus from being called
+          expect(RouterUtil.handleResponseStatus.calls.count()).toBe(0);
```

Comment:
the other way to do this is that we check the ajax calls we make here. There should be only 1, which is the ```previous``` link.

Reply:
Spoke to Mau about this offline. ajax calls may not be accurate here.